static/js/StaticPages.jsx

```diff
@@ -3126,6 +3132,327 @@ const JobsPage = memo(() => {
+        panel_number: 0,   
+        panel_name: "Products",
+        position: rank,
+        text: cta,
+        experiment: label === 'Experiment' ?  1 : 0,
+        feature_name: product,  
+        action: link_type,
+        engagement_type: "navigation",
+        engagement_value: 0
+    });
+}
+
+
+// The call-to-action (link) in the heading of each product
+// TODO - uncomment <OnInView /> once analytics is confirmed
+const ProductCTA = ({product, cta}) => {
+    return (
+
+        // <OnInView onVisible={() => productsAnalytics(product?.rank, product?.titles.en, cta.text.en, product?.type.en, "viewed")}>
+        <a href={cta.url} onClick={productsAnalytics(product.rank, product.titles.en, cta.text.en, product.type.en, "cta", "clicked")}>
+            {cta.icon.url && <img className="productsCTAIcon" 
+                                    src={cta.icon.url}
+                                    alt="Click icon" />}
+                                
+            <span className="productsCTA">
+                <InterfaceText text={{en: cta.text.en , he: cta.text.he }} />
+            </span>
+        </a>
+
+        // </OnInView> 
+
+
+    );
+};
+
+// The main body of each product entry, containing an image and description
+const ProductDesc = ({product}) => {
+    return (
+        <div className="productInner">
+            <div className='productImgWrapper'>
+                <img src={product.rectanglion.url} alt={`Image for product: ${product?.titles?.en}`}/>
+            </div>
+            <div className='productDescWrapper'>
+                <InterfaceText markdown={{en: product.desc?.en, he: product.desc?.he }} />
+            </div>            
+        </div>
+    );
+};
+
+// The main product component, comprised of the building block sub-components
+const Product = ({product}) => {
+    return (
+        <div className="product">
+            <div className="productsHeader">
+                <ProductTitle product={product} />
+                <div className="cta">
+                    {product.ctaLabels?.map(cta => (
+                        <ProductCTA key={cta.id} product={product} cta={cta} />
+                    ))}
+                </div>       
+            </div>
+            <ProductDesc product={product} />
+        </div>
+    );
+};
+
+
+
+
+const ProductsPage = memo(() => {
+    const [products, setProducts] = useState([]);
+    const [error, setError] = useState(null);
+
+    useEffect(() => {
+        loadProducts();
+    }, []);
+
+    // GraphQL query to Strapi
+    const fetchProductsJSON = async () => {
+
+        // If the viewer is an admin, edit the query to retrieve the drafts as well
+        var includeDrafts = '';
+        if (Sefaria.is_moderator){
+            includeDrafts = 'publicationState:PREVIEW,';
+        }
+        
+        const query = `query {
+            products (
+                pagination: { limit: -1 },
+                ${includeDrafts}
+                sort: "rank:asc"
```

Comment:
I think using a relation to create an ordering is better (what I suggested before), because that way you don't need to be aware of the rankings of other products to create a specific ordering. All good though if the ordering does not need to be so clear-cut

Reply:
Hmmm okay. For now, I'll keep it like this - since there was a lot of thought that went into this approach. Good to keep in mind for the future. Thank you!